virttest/utils_backup.py

```diff
@@ -339,10 +339,11 @@ def dump_data_to_file(data_map, source_file_path, target_file_path, driver="qcow
     :param driver: The image format of the source file
     """
     with open(target_file_path, "w"):
-        cmd = "qemu-io -f %s -c \"read -vC %s %s\" %s -r -U  | sed '$d'"
         for entry in data_map:
-            cmd %= (driver, entry["start"], entry["length"], source_file_path)
-            cmd += ">> %s" % target_file_path
+            cmd = (
+                f"qemu-io -f {driver} -c 'read -vC {entry['start']} {entry['length']}'"
```

Comment:
Seems there're more than one pair of single quotes, but looks like it's working well, so it's ok to me.